src/lib/dev-environment/dev-environment-cli.js

```diff
@@ -244,7 +252,7 @@ const componentDisplayNames = {
 	clientCode: 'site-code',
 };
 
-export async function promptForComponent( component: string, allowLocal: boolean, defaultObject: ComponentConfig ): Promise<ComponentConfig> {
+export async function promptForComponent( component: string, allowLocal: boolean, defaultObject: ComponentConfig | null ): Promise<ComponentConfig> {
```

Comment:
We may pass `null` here in https://github.com/Automattic/vip/blob/c6c7ee04be649d965f666cdab7b0e680bdd3d271/src/lib/dev-environment/dev-environment-cli.js#L192, https://github.com/Automattic/vip/blob/c6c7ee04be649d965f666cdab7b0e680bdd3d271/src/lib/dev-environment/dev-environment-cli.js#L207

Cannot call `promptForComponent` with `defaultObject` bound to `defaultObject` because  null [1] is incompatible with  `ComponentConfig` [2].